spec/spec_helper.rb

```diff
@@ -160,10 +160,14 @@ module PuppetSpec
   PUPPET_FACTER_2_GCE_URL = %r{^http://metadata/computeMetadata/v1(beta1)?}.freeze
   PUPPET_FACTER_3_GCE_URL = "http://metadata.google.internal/computeMetadata/v1/?recursive=true&alt=json".freeze
 
+  # Facter azure metadata endpoint
+  PUPPET_FACTER_AZ_URL = "http://169.254.169.254/metadata/instance?api-version=2020-09-01"
+
   config.around :each do |example|
     # Ignore requests from Facter GCE fact in Travis
```

Comment:
this comment is a bit outdated now, would be good to update it

Reply:
also I'm not sure if we need to stub GCE anymore, or maybe we do...

Reply:
`fact_value = bios_vendor&.include?('Google') ? Facter::Resolvers::Gce.resolve(:metadata) : nil`

Reply:
I'm for stubbing all the calls to external services, don't want to make assumptions form where people are running the tests :D

Reply:
should we also stub the EC2 endpoints then?

Reply:
Added also EC2